sdk/cognitivelanguage/ai-language-text/src/lro.ts

```diff
@@ -266,12 +266,31 @@ type Writable<T> = {
  */
 export function createUpdateAnalyzeState(docIds?: string[]) {
   return (state: AnalyzeBatchOperationState, lastResponse: LroResponse): void => {
-    const { createdOn, modifiedOn, id, displayName, expiresOn, tasks, lastUpdateDateTime } =
-      lastResponse.flatResponse as AnalyzeTextJobStatusResponse & { lastUpdateDateTime: string };
+    const {
+      createdOn,
+      modifiedOn,
+      id,
+      displayName,
+      expiresOn,
+      tasks,
+      lastUpdateDateTime,
+      status,
+      errors,
+    } = lastResponse.flatResponse as AnalyzeTextJobStatusResponse & { lastUpdateDateTime: string };
     const mutableState = state as Writable<AnalyzeBatchOperationState> & {
       docIds?: string[];
     };
     mutableState.createdOn = createdOn;
+    if (status.toLowerCase() === "partiallysucceeded") {
+      if (!errors) {
+        mutableState.status = "succeeded";
+      } else {
+        const errorString = errors.map(({ message }) => message).join("\n");
+        const error = new Error(errorString);
+        mutableState.status = "failed";
+        mutableState.error = error;
```

Comment:
nitpick: this could be simplified to:
```suggestion
        mutableState.status = "failed";
        mutableState.error = new Error(errorString);
```